go/cmd/dolt/commands/clone.go

```diff
@@ -228,3 +232,14 @@ func validateAndParseDolthubUrl(urlStr string) (string, bool) {
 
 	return "", false
 }
+
+func getUserAndPassConfig(apr *argparser.ArgParseResults) *creds.DoltCredsForPass {
```

Comment:
I see we will support this for `clone` for now, with `pull/fetch` to come later probably?

I think I would make this error --- if `apr.Contains(cli.UserParam)` but `os.Getenv("DOLT_REMOTE_PASSWORD") == nil`, the user's environment is probably setup incorrectly and we could give them a more useful error than just an auth failure against the remote.